scipy/special/_basic.py

```diff
@@ -2414,13 +2415,19 @@ def factorial2(n, exact=False):
         Double factorial of `n`, as an int or a float depending on
         `exact`.
 
+    References
+    ----------
+    ..[1] https://functions.wolfram.com/GammaBetaErf/Factorial2/27/01/0002/
+
     Examples
     --------
     >>> from scipy.special import factorial2
     >>> factorial2(7, exact=False)
     array(105.00000000000001)
     >>> factorial2(7, exact=True)
     105
+    >>> factorial2(1j)
+    array(2.7171303558030684+0.2795273169425987j)
 
     """
     if exact:
```

Comment:
Looking closer, the `exact` case definitely needs to be guarded against non-integral inputs.

Reply:
Correct me if I am wrong but is this not what I have done in  #15299?

Reply:
> Correct me if I am wrong but is this not what I have done in #15299?

You're right; I didn't follow the link for "Somewhat related to #15299", sorry...

I'd upgrade this to a "Blocked on #15299", because without those checks, this PR would lead to an IMO broken state of affairs.

Reply:
Agreed